### PR TITLE
Fix nrf_cloud bootloader FOTA handling

### PR DESCRIPTION
**Pull request by jayteemo**
The nrf_cloud library was not properly processing bootloader FOTA updates.
Updated to reboot and check active slot so that the FOTA status is correctly reported to nRF Cloud.
Also updated the REST FOTA sample and ATv2 as a result of FOTA testing.

CIA-554

### CODE REVIEW

File: subsys/net/lib/fota_download/src/fota_download.c [hunk 1]
```diff
@@ -300,6 +300,42 @@ static bool is_ip_address(const char *host)
...
+		return -EFAULT;
+	}
+
+	s1 = fw_info_find(PM_S1_ADDRESS);
+	if (s1 == NULL) {
+		/* No s1 found, s0 is active */
+		*s0_active = true;
+	} else {
+		/* Both s0 and s1 found, check who is active */
+		*s0_active = s0->version >= s1->version;
```
Comment (plskeggs) [inline subsys/net/lib/fota_download/src/fota_download.c:330]: I'm suspicious of this.  How do you know that s0 is active and not s1, if the version is equal?

File: applications/asset_tracker_v2/src/main.c [hunk 2]
```diff
@@ -534,7 +534,7 @@ void main(void)
 	int err;
 	struct app_msg_data msg;
 
-	if (!IS_ENABLED(CONFIG_LWM2M_CARRIER)) {
+	if (!IS_ENABLED(CONFIG_LWM2M_CARRIER) && !IS_ENABLED(CONFIG_NRF_CLOUD_FOTA)) {
```
Comment (plskeggs) [inline applications/asset_tracker_v2/src/main.c:537]: Just curious -- why was this change needed?

**Review by plskeggs — APPROVED** [review]

File: include/net/fota_download.h [hunk 3]
```diff
@@ -161,6 +161,15 @@ int fota_download_cancel(void);
...
 int fota_download_target(void);
 
+/**@brief Get the active bootloader (B1) slot, s0 or s1.
+ *
+ * @param[out] s0_active Set to 'true' if s0 is active slot, 'false' otherwise
+ *
+ * @retval 0 If successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int fota_download_s0_active_get(bool *const s0_active);
```
[the remark this answers is not in the document]
Comment (jayteemo) [reply include/net/fota_download.h:171]: i have no preference...  i named it that since there is a similar function `spm_s0_active`.

File: applications/asset_tracker_v2/src/main.c (hunk 2 above)
Comment (jayteemo) [reply applications/asset_tracker_v2/src/main.c:537]: without this change, `handle_nrf_modem_lib_init_ret` will read the modem lib init value and reboot before the nrf_cloud lib is able process the value.  this results in an unvalidated fota result on nrf cloud.

File: subsys/net/lib/fota_download/src/fota_download.c (hunk 1 above)
Comment (jayteemo) [reply subsys/net/lib/fota_download/src/fota_download.c:330]: that is how the bootloader handles the slots/versions:
https://github.com/nrfconnect/sdk-nrf/blob/bc2e9b452cb49093eff50e25a8fc2a9d8ce63a9a/samples/bootloader/src/main.c#L95

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -201,6 +211,22 @@ Libraries for networking
...
+
+* :ref:`lib_nrf_cloud` library:
+
+  * Added:
+
+    * :c:func:`nrf_cloud_bootloader_fota_slot_set` function that sets the active bootloader slot flag during bootloader FOTA updates.
+    * :c:func:`nrf_cloud_pending_fota_job_process` function that processes the state of pending FOTA jobs.
+
+  * Fixed:
+    * Validation of bootloader FOTA updates.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:228]:
```suggestion
  * Fixed:
  
    * Validation of bootloader FOTA updates.
```
Just for consistency, added an empty line. Should work fine without as well.

**Review by peknis — CHANGES_REQUESTED** [review]

File: applications/asset_tracker_v2/src/main.c (hunk 2 above)
Comment (plskeggs) [reply applications/asset_tracker_v2/src/main.c:537]: Good catch.

File: include/net/fota_download.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (plskeggs) [reply include/net/fota_download.h:171]: OK.

**Review by peknis — APPROVED** [review]

**Review by rlubos — APPROVED** [review]
